lib/miq_automation_engine/engine/miq_ae_builtin_method.rb

```diff
@@ -160,5 +171,28 @@ def self.category_for_key(obj, key)
       end
     end
     private_class_method :category_for_key
+
+    def self.prepend_vendor(obj)
+      vendor = nil
+      ATTRIBUTE_LIST.detect { |attr| vendor = detect_vendor(obj.workspace.root[attr], attr) }
+      if vendor
+        $miq_ae_logger.info("Setting prepend_namespace to: #{vendor}")
+        obj.workspace.prepend_namespace = vendor
+      end
+    end
+    private_class_method :prepend_vendor
+
+    def self.detect_vendor(src_obj, attr)
+      case attr
+      when "orchestration_stack"
+        src_obj.type.split('::')[2] if src_obj
+      when "miq_host_provision"
+        "vmware" if src_obj
+      when "miq_request", "miq_provision", "vm_migrate_task"
+        src_obj.source.vendor if src_obj && src_obj.source.respond_to?(:vendor)
+      when "vm"
+        src_obj.vendor if src_obj && src_obj.respond_to?(:vendor)
+      end
+    end
```

Comment:
Yes, that is what I was thinking.  We should be able to ask the object instead of tying to figure out how to extract that information here.  For example, the `orchestration_stack` logic is very fragile and will fail if/when someone implements a different namespace structure for their models.